src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -130,6 +131,16 @@ private void OnRpcDisconnected(object sender, JsonRpcDisconnectedEventArgs e)
                     var descriptor = new ServiceDescriptor(serviceName) { HostGroup = hostGroup };
                     return await client.RequestServiceAsync(descriptor, cancellationToken).ConfigureAwait(false);
                 }
+                catch (OperationCanceledException)
+                {
+                    // if it is our own cancellation token, then rethrow
+                    // otherwise, let us retry.
```

Comment:
I am fine we doing retry. but this timeout was very hidden so hard to find out what is going on until I dig through and read service hub source code.

I think if timeout is part of contract, it should accept timeout value or behavior flag and/or should throw timeout exception rather than cancellation exception.